FirebaseMessaging/Sources/Token/FIRMessagingTokenManager.m

```diff
@@ -180,6 +180,18 @@ - (void)tokenWithAuthorizedEntity:(NSString *)authorizedEntity
     [tokenOptions addEntriesFromDictionary:options];
   }
 
+  //ensure we have an APNS Token
+  if (tokenOptions[kFIRMessagingTokenOptionsAPNSKey] == nil) {
```

Comment:
Have you tested this with fresh install? I believe the first request is almost always not including a APNS token. 

Would this show as a critical error to developers or just warning  on the API surface level? I want to make sure they are not thinking things stop working even though the second request will go through

Reply:
I tested three different scenarios (also included in PR description) to confirm we don't regress - 

- Fresh install of app with keychain cleared.
- Reinstall of app via Xcode
- Fresh install of app with keychain not cleared.

For each scenario, confirmed that a notification was sent as expected to the vended FCM Token.

It gets 
- logged as a WARNING. I am on the fence if this should be logged as ERROR. 
- the completion block returns an NSError with a new code kFIRMessagingErrorCodeMissingDeviceToken

Reply:
Also added a unit test that checks for an NSError when device token is missing.

Reply:
Thanks for explaining! I think it should be warning because apple might return the apns token with a delay then the first token request will fail.
Unless we changed the logic that only send the first token request when apple returns apns token. I don't remember if we made the change yet.

Reply:
Ok cool - I'll leave it as WARN. 

>>Unless we changed the logic that only send the first token request when apple returns apns token. I don't remember if we made the change yet.

The actual flow (I think its part of swizzling logic) hasn't been changed here - but we won't make a register call if there is no APNS token.